app/Http/Controllers/UserController.php

```
@@ -234,20 +224,20 @@ public function update(Request $request, User $user)
-            $user->syncRoles([$roleName]);
-        }
+        $user->syncRoles([$role]);
 
-        // Si el usuario es encargado y se le asigna una sucursal, actualizar el manager_id de la sucursal
-        if ($validated['role'] === 'encargado' && isset($validated['branch_id'])) {
+        // Igual que en store(): solo el rol Encargado del sistema dispara el
+        // auto-registro como gerente de sucursal.
+        $isSystemEncargado = $role->is_system && $role->name === DefaultRoleProvisioner::ENCARGADO;
+        if ($isSystemEncargado && isset($validated['branch_id'])) {
             // Si el usuario ya era encargado de otra sucursal, quitarlo como encargado
             Branch::where('manager_id', $user->id)
```

PHPStan (2.2.8)

[error] Call to an undefined static method App\Models\Branch::where().


(staticMethod.notFound)